webapp/src/tutorialCodeValidation.tsx

```diff
@@ -60,42 +60,52 @@ export class ShowValidationMessage extends data.Component<TutorialCodeValidation
             const prev = this.props.ruleComponents.find(r => r.ruleName == rule.ruleName);
             // Update if the list of blockids is different
             if ((rule.blockIds && rule.blockIds.sort().toString() != prev?.blockIds?.sort()?.toString())
-                || rule.blockIds.length != nextState.ruleSnippets?.[rule.ruleName]?.length) {
-                this.getRuleSnippetsAsync(rule);
+                || rule.blockIds.length != nextState.ruleBlocks?.[rule.ruleName]?.length) {
+                this.getRuleBlocksAsync(rule);
             }
         })
     }
 
     renderRule(rule: pxt.tutorial.TutorialRuleStatus, index?: number) {
-        const snippets = this.state.ruleSnippets?.[rule.ruleName];
-        if (!snippets) {
+        const blockUris = this.state.ruleBlocks?.[rule.ruleName];
+        if (!blockUris) {
             return <p key={index + rule.ruleName}>{(rule.ruleTurnOn && !rule.ruleStatus) ? rule.ruleMessage : ''}</p>
         } else {
-            // TODO (jxwoon): render snippets
+            // TODO (jxwoon): render block dataUris
             return <p key={index + rule.ruleName}>{(rule.ruleTurnOn && !rule.ruleStatus) ? rule.ruleMessage : ''}</p>
         }
     }
 
-    getRuleSnippetsAsync(rule: pxt.tutorial.TutorialRuleStatus) {
+    getRuleBlocksAsync(rule: pxt.tutorial.TutorialRuleStatus) {
```

Comment:
not in this pr; this seems like it might be a good function to async-ify to clean up eventually